pkgs/top-level/all-packages.nix

```diff
@@ -15462,6 +15462,8 @@ with pkgs;
   };
   qbittorrent-nox = qbittorrent.override { guiSupport = false; };
 
+  qbittorrent-enhanced-nox = callPackage ../by-name/qb/qbittorrent-enhanced/package.nix { guiSupport = false; };
```

Comment:
```suggestion
  qbittorrent-enhanced-nox = qbittorrent-enhanced.override { guiSupport = false; };
```

Reply:
Can the name of the binary be overridden as well? since even tho the application is qbittorrent-enhanced, the binary is just qbittorrent, making it fail on macos.
Or should I change the name from postInstall and mainProgram to just qbittorrent?

Reply:
`.override` should also work.

But I wonder what's the different between `.override` and `callPackage .. { }`, is it related to package splicing?

I've seen quite a lot of `callPackage ../by-name` usages:

```
$ grep -F 'callPackage ../by-name' pkgs/top-level/all-packages.nix | wc -l
147
```

But `.override` with packages from `by-name` can not be easily counted.